src/main/java/com/mraof/minestuck/client/gui/DialogueScreen.java

```diff
@@ -106,16 +176,35 @@ public void render(GuiGraphics guiGraphics, int mouseX, int mouseY, float partia
 		
 		guiGraphics.blit(dialogueData.guiBackground(), xOffset, yOffset, 0, 0, GUI_WIDTH, GUI_HEIGHT);
 		
+		renderAnimation(guiGraphics);
+		
 		int pY = yOffset + 20;
 		for(FormattedCharSequence line : messageLines)
 		{
 			guiGraphics.drawString(font, line, xOffset + 10, pY, 0x000000, false);
-			pY += 9;
+			pY += DialogueButton.TEXT_SPACING;
 		}
 		
 		super.render(guiGraphics, mouseX, mouseY, partialTicks);
 	}
 	
+	/**
+	 * Places a talksprite directly to the right of the gui. The center height of the sprite is level with that of the gui.
+	 */
+	private void renderAnimation(GuiGraphics guiGraphics)
+	{
+		ResourceLocation sprite = animationData.getRenderPath(dialogueData.spriteType());
+		float scale = animationData.scale();
+		int moddedWidth = (int) (animationData.spriteWidth() * scale);
+		int moddedHeight = (int) (animationData.spriteHeight() * scale);
+		
+		//if there is a .png.mcmeta file associated with the sprite, the animation for it is updated here
+		AnimatableTexture.setAndUpdate(sprite, animationTick);
+		guiGraphics.blit(sprite, xOffset + GUI_WIDTH + animationData.xOffset(), (this.height / 2) - (animationData.spriteHeight() / 2), 0, 0, moddedWidth, moddedHeight, moddedWidth, moddedHeight);
+		
+		animationTick++;
```

Comment:
```suggestion
		AnimatableTexture.setAndUpdate(sprite, animationTick++);
		guiGraphics.blit(sprite, xOffset + GUI_WIDTH + animationData.xOffset(), (this.height / 2) - (animationData.spriteHeight() / 2), 0, 0, moddedWidth, moddedHeight, moddedWidth, moddedHeight);
```
Not required, but I thought this would be nice since the increment is very much a part of updating the animation.